pkg/pipeline/decode/decode_protobuf.go

```diff
@@ -37,26 +37,35 @@ func PBFlowToMap(flow *pbflow.Record) config.GenericMap {
 		return config.GenericMap{}
 	}
 	out := config.GenericMap{
-		"FlowDirection":   int(flow.Direction.Number()),
-		"Bytes":           flow.Bytes,
-		"SrcAddr":         ipToStr(flow.Network.GetSrcAddr()),
-		"DstAddr":         ipToStr(flow.Network.GetDstAddr()),
-		"SrcMac":          macToStr(flow.DataLink.GetSrcMac()),
-		"DstMac":          macToStr(flow.DataLink.GetDstMac()),
-		"SrcPort":         flow.Transport.GetSrcPort(),
-		"DstPort":         flow.Transport.GetDstPort(),
-		"Etype":           flow.EthProtocol,
-		"Packets":         flow.Packets,
-		"Duplicate":       flow.Duplicate,
-		"Proto":           flow.Transport.GetProtocol(),
-		"TimeFlowStartMs": flow.TimeFlowStart.AsTime().UnixMilli(),
-		"TimeFlowEndMs":   flow.TimeFlowEnd.AsTime().UnixMilli(),
-		"TimeReceived":    time.Now().Unix(),
-		"Interface":       flow.Interface,
-		"AgentIP":         ipToStr(flow.AgentIp),
-		"Flags":           flow.Flags,
-		"IcmpType":        flow.Icmp.GetIcmpType(),
-		"IcmpCode":        flow.Icmp.GetIcmpCode(),
+		"FlowDirection":     int(flow.Direction.Number()),
+		"Bytes":             flow.Bytes,
+		"SrcAddr":           ipToStr(flow.Network.GetSrcAddr()),
+		"DstAddr":           ipToStr(flow.Network.GetDstAddr()),
+		"SrcMac":            macToStr(flow.DataLink.GetSrcMac()),
+		"DstMac":            macToStr(flow.DataLink.GetDstMac()),
+		"SrcPort":           flow.Transport.GetSrcPort(),
+		"DstPort":           flow.Transport.GetDstPort(),
+		"Etype":             flow.EthProtocol,
+		"Packets":           flow.Packets,
+		"Duplicate":         flow.Duplicate,
+		"Proto":             flow.Transport.GetProtocol(),
+		"TimeFlowStartMs":   flow.TimeFlowStart.AsTime().UnixMilli(),
+		"TimeFlowEndMs":     flow.TimeFlowEnd.AsTime().UnixMilli(),
+		"TimeReceived":      time.Now().Unix(),
+		"Interface":         flow.Interface,
+		"AgentIP":           ipToStr(flow.AgentIp),
+		"Flags":             flow.Flags,
+		"IcmpType":          flow.Icmp.GetIcmpType(),
+		"IcmpCode":          flow.Icmp.GetIcmpCode(),
+		"TcpDropBytes":      flow.TcpDropBytes,
+		"TcpDropPackets":    flow.TcpDropPackets,
+		"TcpDropFlags":      flow.GetTcpDropFlags(),
+		"TcpDropState":      tcpStateToStr(flow.GetTcpDropState()),
+		"TcpDropCause":      tcpDropCauseToStr(flow.GetTcpDropCause()),
```

Comment:
:+1: that's perfect to centralize the code ! 

Can we also do that with ICMP Type & Code fields above ?
Should we also do that for Protocol ?